src/emulator/util/src/util.cpp

```diff
@@ -170,6 +170,13 @@ std::u16string utf8_to_utf16(const std::string &str) {
 }
 #endif
 
+std::string toupper(std::string s) {
```

Comment:
```suggestion
std::string toupper(const std::string &s) {
```

Passing it as reference, since you did make a copy of ``s`` below

Reply:
Do same for header